airflow/providers/amazon/aws/hooks/glue.py

```diff
@@ -0,0 +1,214 @@
+    :type str
+    :param iam_role_name: AWS IAM Role for Glue Job
+    :type str
+    """
+
+    def __init__(self,
+                 job_name=None,
+                 desc=None,
+                 concurrent_run_limit=None,
+                 script_location=None,
+                 conns=None,
+                 retry_limit=None,
+                 num_of_dpus=None,
+                 aws_conn_id='aws_default',
+                 region_name=None,
+                 iam_role_name=None,
+                 s3_bucket=None, *args, **kwargs):
+        self.job_name = job_name
+        self.desc = desc
+        self.concurrent_run_limit = concurrent_run_limit or 1
+        self.script_location = script_location
+        self.conns = conns or ["s3"]
+        self.retry_limit = retry_limit or 0
+        self.num_of_dpus = num_of_dpus or 10
+        self.aws_conn_id = aws_conn_id
+        self.region_name = region_name
+        self.s3_bucket = s3_bucket
+        self.role_name = iam_role_name
+        self.S3_PROTOCOL = "s3://"
+        self.S3_ARTIFACTS_PREFIX = 'artifacts/glue-scripts/'
+        self.S3_GLUE_LOGS = 'logs/glue-logs/'
+        super(AwsGlueJobHook, self).__init__(*args, **kwargs)
+
+    def get_conn(self):
+        conn = self.get_client_type('glue', self.region_name)
+        return conn
+
+    def list_jobs(self):
+        conn = self.get_conn()
+        return conn.get_jobs()
+
+    def get_iam_execution_role(self):
+        """
+        :return: iam role for job execution
+        """
+        iam_client = self.get_client_type('iam', self.region_name)
+
+        try:
+            glue_execution_role = iam_client.get_role(RoleName=self.role_name)
+            self.log.info("Iam Role Name: {}".format(self.role_name))
+            return glue_execution_role
+        except Exception as general_error:
+            raise AirflowException(
+                'Failed to create aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
+
+    def initialize_job(self, script_arguments=None):
+        """
+        Initializes connection with AWS Glue
+        to run job
+        :return:
+        """
+        if self.s3_bucket is None:
+            raise AirflowException(
+                'Could not initialize glue job, '
+                'error: Specify Parameter `s3_bucket`'
+            )
+
+        glue_client = self.get_conn()
+
+        try:
+            job_response = self.get_or_create_glue_job()
+            job_name = job_response
+            job_run = glue_client.start_job_run(
+                JobName=job_name,
+                Arguments=script_arguments
+            )
+            return self.job_completion(job_name, job_run['JobRunId'])
+        except Exception as general_error:
+            raise AirflowException(
+                'Failed to run aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
```

Comment:
Same here. Can't you narrow down the Exception type? Do you really want to except every type?